snuba/migrations/policies.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from snuba.migrations.migration import Migration
+from snuba.migrations.status import Status
+
+
+class MigrationAction(Enum):
+    FORWARDS = "forwards"
+    BACKWARDS = "backwards"
+
+
+class MigrationPolicy(ABC):
+    """
+    A MigrationPolicy implements an `is_allowed` method
+    that determines whether or not a migration can be applied
+    from snuba admin.
+
+    A policy can be used by a migration group to determine the
+    level of access a group has to run migrations.
+    """
+
+    @abstractmethod
+    def allows(migration: Migration, action: MigrationAction) -> bool:
+        raise NotImplementedError
```

Comment:
How about
```suggestion
    def can_run(migration: Migration) -> bool:
        raise NotImplementedError
       
     def can_reverse(migration: Migration) -> bool:
        raise NotImplementedError
```
Since there's only two options the enum doesn't feel all that necessary. `allows` is a bit vague as it doesn't specify what you're being allowed to do

Reply:
Is there  case we would want someone to have access to only one of these actions? Why split the access between run/reverse?

Reply:
@lynnagara yeah could make sense to do. Also I think it might make sense to pass the `migration_key` instead of migration since getting the migration itself is not aware of it's own status.